src/NLog/Common/InternalLogger.cs

```diff
@@ -320,6 +320,7 @@ private static void Write(LogLevel level, string message, object[] args)
                 var logFile = LogFile;
                 if (!string.IsNullOrEmpty(logFile))
                 {
+                    CreateDirectoriesIfNeeded(logFile);
```

Comment:
Unfortunately, that will make my unit test moot, but agree with the performance.